view/frontend/web/template/payment/offsite-truemoney-form.html

```diff
@@ -0,0 +1,61 @@
+<div class="payment-method"
+    data-bind="css: {'_active': (getCode() == isChecked())}">
+    <div class="payment-method-title field choice">
+        <input type="radio" name="payment[method]" class="radio"
+            data-bind="attr: {
+               'id': getCode()
+               },
+               value: getCode(),
+               checked: isChecked,
+               click: selectPaymentMethod,
+               visible: isRadioButtonVisible(),
+               enable: isActive()" />
+        <label data-bind="attr: {'for': getCode()}" class="label">
+            <span data-bind="text: getTitle()"></span>
+        </label>
+        <div data-bind="visible: isSandboxOn()" class="page messages">
```

Comment:
This is definitely in the wrong place, we shouldn't be adding another warning for every new payment method. Test/Sandbox mode always applies to all Omise payment methods. The warning should really be a single one elsewhere on the page. If we really do want to put a warning on every payment method though, we should at least be calling the same "isSandbox" method to decide whether or not to display it. All this code duplication is bad.

I guess this relates back to the other refactoring we discussed before.

Reply:
what if a merchant has PayPal on or cash on delivery. We should show exactly which payments are in test mode.

Reply:
The message is only relevant for the merchant. Telling them that 'Omise payment methods' are in test mode is definitely enough info. We should credit the merchant with some intelligence

Reply:
It's about reminding them that test mode is on, so they don't leave it on when going live

Reply:
then if it is the only a reminder, that is fine with the current implementation, 

and also it covers the situation when a merchant will really forget. The current implementation is way better than showing a single message.

Reply:
The current implementation is overly complex, adds a maintenance burden, and treats the merchant like an idiot by telling them again and again and again

Reply:
your aggressiveness in public or private debate is something you should consider.
if you feel hard to review PRs in a civilised way I will stop sending you requests.

Reply:
No aggression was intended. It was merely a statement of fact. No capital letters, no bashing of keyboard 🌻

Reply:
I agreed on showing one message somewhere is better than showing in every payment methods. Just for 2 main reasons.

1. Complexity of the code, as this case may be a good example that we can easily forget about adding this label, also the code has been implemented everywhere, it's hard to maintenance and risk of human error. 

2. As user experience, having "test mode" in each payment methods may raise a question:
"Is it possible to turn `Alipay` to test mode while keep the rest in live mode." As the design is indicating that "test mode" is under and controlled by each individual payment method.

If we talking about achieving on raising an awareness of "accidentally leaving the test mode on", I think both are achieving the same goal, but having one place to maintain would give more benefit than another.

Reply:
However, do we have any conclusion on this? : )))
What action should we take? Want to fix this one before this PR? or we may submit another PR to improve this code, maybe?

Reply:
Another thing that occurred to me over the weekend was that putting the warnings actually next to the payment methods like this will mean that they only see the warnings once they have gone through all the previous checkout steps. The likelihood of them leaving test mode on in this case is for that reason probably higher than if the warning is outside of this area and always visible

Reply:
🤚 One thing that not related to the above but I just realised. 
If nothing has changed, I think we should stick with `isTest()` (or `isTestMode()`) rather than `isSandboxOn`. As normally Omise uses `Test` term widely and almost everywhere.

Reply:
Thanks for all comments, but for now, it stays at it is.

Reply:
@jonrandy good point. 👍

Reply:
> I agreed on showing one message somewhere is better than showing in every payment methods. Just for 2 main reasons.
> 
>     1. Complexity of the code, as this case may be a good example that we can easily forget about adding this label, also the code has been implemented everywhere, it's hard to maintenance and risk of human error.
> 
>     2. As user experience, having "test mode" in each payment methods may raise a question:
>        "Is it possible to turn `Alipay` to test mode while keep the rest in live mode." As the design is indicating that "test mode" is under and controlled by each individual payment method.
> 
> 
> If we talking about achieving on raising an awareness of "accidentally leaving the test mode on", I think both are achieving the same goal, but having one place to maintain would give more benefit than another.

Totally agree on 1) - the human error thing already almost happened as the warning was initially forgotten for True Money. As for 2) I agree also - having a warning for every method does imply that it may be possible to put each method in test mode individually

Reply:
@guzzilar yes.. your function name might be better.
but it is consistent with what we have in the admin panel.
we use `Sandbox`
I used to display here `Sandbox` too, but it was changed by support team. this message.
that is why is sandbox: 
![image](https://user-images.githubusercontent.com/10651523/60423200-eef37d80-9bed-11e9-975e-43e24d54e1fd.png)
 We could consider changing all of that to testmode including panel info.

Reply:
for somebody who work with the code it might be easier to recognize what exactly is he dealing with when we leave `sandbox`

text messages are always fully customizable,
in polish, we would use `sandbox` for instance if you would translate it.
not test mode.

Reply:
Just to make entire Omise products consistent (test key, test account, test mode, as we mentioned in documents and Dashboard) I still vote that we should consider replace all `Sandbox` to `Test mode` but in the future PR. ;)

Reply:
ok @guzzilar this cycle will do refactoring.